flyteadmin/pkg/manager/impl/execution_manager.go

```diff
@@ -2050,6 +2055,41 @@ func (m *ExecutionManager) addProjectLabels(ctx context.Context, projectName str
 	return initialLabels, nil
 }
 
+// addUserAnnotations automatically injects user identity information as annotations when enabled in config.
+// This allows tracking which user submitted each workflow execution and enables user-based authorization.
+func (m *ExecutionManager) addUserAnnotations(ctx context.Context, initialAnnotations map[string]string) map[string]string {
+	// Check if user annotation injection is enabled
+	if !m.config.ApplicationConfiguration().GetTopLevelConfig().GetInjectUserAnnotations() {
+		return initialAnnotations
+	}
+
+	// Get user identity from authentication context
+	identityContext := auth.IdentityContextFromContext(ctx)
+	var principal string
+	if identityContext.UserInfo() != nil {
+		principal = identityContext.UserInfo().GetEmail()
+	}
+
+	if principal == "" {
+		// If no email is available, skip annotation injection
+		logger.Debugf(ctx, "No user email found in context, skipping user annotation injection")
+		return initialAnnotations
+	}
+
+	if initialAnnotations == nil {
+		initialAnnotations = make(map[string]string)
+	}
+
+	prefix := m.config.ApplicationConfiguration().GetTopLevelConfig().GetUserAnnotationPrefix()
+	userKey := prefix + "/user"
```

Comment:
Make this configurable as i suggest above and would be great if we can differentiate between a app user and regular user